automation/scripts/clean_pipelines/clean_pipelines.py

```diff
@@ -0,0 +1,392 @@
+# Copyright 2019 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+import typing
+from time import sleep
+
+import mlrun_pipelines
+import pandas as pd
+import pytz
+from dateutil import parser
+from kfp import Client
+from mlrun_pipelines.models import PipelineRun
+
+import mlrun
+
+
+def validate_and_convert_date(date_input: str) -> str:
+    """
+    Converts any recognizable date string into a standardized RFC 3339 format.
+    :param date_input: A date string in a recognizable format.
+    """
+    try:
+        dt_object = parser.parse(date_input)
+        if dt_object.tzinfo is not None:
+            # Convert to UTC if it's in a different timezone
+            dt_object = dt_object.astimezone(pytz.utc)
+        else:
+            # If no timezone info is present, assume it's in local time
+            local_tz = pytz.timezone("UTC")
+            dt_object = local_tz.localize(dt_object)
+
+        formatted_date = dt_object.isoformat().replace("+00:00", "Z")
+        if not formatted_date.endswith("Z"):
+            formatted_date += "Z"
+
+        return formatted_date
+    except (ValueError, OverflowError) as e:
+        raise ValueError(
+            f"Invalid date format: {date_input}. Please provide a valid date."
+        ) from e
+
+
+def get_kfp_client(
+    kfp_url=mlrun.mlconf.kfp_url, namespace: str = "default-tenant"
+) -> Client:
+    kfp_client = mlrun_pipelines.utils.get_client(kfp_url, namespace)
+    return kfp_client
```

Comment:
you can possibly use `mlrun.mlconf.resolve_namespace()` and don't use `default-tenant` because it's not a thing in CE